### PR TITLE
Adds toleration for arm and amd to operator components

### PR DESCRIPTION
**Pull request by meik99**
# Description

Adds tolerations to operator components so that they are deployed on ARM nodes as well

## How can this be tested?

* Acquire an ARM cluster
* Generate the manifest
* Deploy the manifest (an arm image with the snapshot-<branch> tag exists already)
* Observe

## Checklist
- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: config/helm/chart/default/values.yaml [hunk 1]
```diff
@@ -21,7 +21,13 @@ installCRD: false
 
 operator:
   nodeSelector: {}
-  tolerations: []
+  tolerations:
```
Comment (chrismuellner) [inline config/helm/chart/default/values.yaml:24]: these tolerations should be unconfigurable and directly set in the deployment/daemonset, so tolerations that are set here are just added to the default tolerations and don't overwrite them.

**Review by chrismuellner — CHANGES_REQUESTED** [review]

File: config/helm/chart/default/values.yaml (hunk 1 above)
Comment (meik99) [reply config/helm/chart/default/values.yaml]: fixed

File: config/helm/chart/default/templates/Common/csi/daemonset.yaml [hunk 2]
```diff
@@ -285,9 +285,16 @@ spec:
...
       nodeSelector: {{- toYaml .Values.csidriver.nodeSelector | nindent 8 }}
       {{- end }}
-      {{- if .Values.csidriver.tolerations }}
-      tolerations: {{- toYaml .Values.csidriver.tolerations | nindent 8 }}
-      {{- end }}
+      tolerations:
+        {{- if .Values.csidriver.tolerations }}
+        {{- toYaml .Values.csidriver.tolerations | nindent 8 }}
+        {{- end }}
+        - key: kubernetes.io/arch
```
Comment (0sewa0) [inline config/helm/chart/default/templates/Common/csi/daemonset.yaml:292]:
```
    - effect: NoSchedule
      key: ToBeDeletedByClusterAutoscaler
      operator: Exists
```
maybe having this here as well would make sense ? (its in the values.yaml, the other 2 in the values.yaml should be configurable)

Comment (meik99) [reply config/helm/chart/default/templates/Common/csi/daemonset.yaml:292]: done

File: hack/make/manifests/manifests.mk [hunk 3]
```diff
@@ -1,5 +1,9 @@
 manifests/prepare-directory:
 	find $(MANIFESTS_DIR) -type f -not -name 'kustomization.yaml' -delete
 
-# Generates manifests e.g. CRD, RBAC etc, for Kubernetes and OpenShift
+## Generates manifests e.g. CRD, RBAC etc, for Kubernetes and OpenShift
 manifests: manifests/prepare-directory manifests/kubernetes manifests/openshift
+
+## Generate manifests for the branch's image tag
+manifests/branch: export MASTER_IMAGE=quay.io/dynatrace/dynatrace-operator:snapshot-${SNAPSHOT_SUFFIX}
```
Comment (0sewa0) [inline hack/make/manifests/manifests.mk:8]: Just out of curiosity, why ?

**Review by 0sewa0 — APPROVED** [review]

File: hack/make/manifests/manifests.mk (hunk 3 above)
Comment (meik99) [reply hack/make/manifests/manifests.mk:8]: I am lazy

Comment (meik99) [reply hack/make/manifests/manifests.mk:8]: the cloudnative tests use the manifests to deploy the operator. The image change is done in the kustomize, by default, the manifests only have the `:snapshot` tag. I did this so I can run the e2e tests more easily

**Pull request merged**